test/integration/adoption_application_reviews_test.rb

```diff
@@ -113,4 +113,24 @@ class AdoptionApplicationReviewsTest < ActionDispatch::IntegrationTest
     follow_redirect!
     assert_select 'a', 'Create Adoption'
   end
+
+  test "after making the http request to create an adoption, the application disappears" do
+    sign_in users(:user_two)
+
+    put "/adopter_applications/#{@adopter_application.id}",
+      params: { adopter_application:
+        {
+          status: 'successful_applicant', notes: ''
+        }, commit: 'Save', id: @adopter_application.id
+      }
+
+    post '/create_adoption',
+      params: { adopter_account_id: adopter_accounts(:adopter_account_one).id, dog_id: dogs(:dog_five).id }
+
+    follow_redirect!
```

Comment:
We should maybe be explicit here and use `get '/adopter_applications'` instead of `follow_redirect!`?

Reply:
For this integration test we don't need to make the PUT request to change status to successful applicant, and then make the POST request. We can just make the POST request alone.

It would be good to prove a difference, i.e. we can GET /adopter_applications then assert that the dog's name is there, then make the POST request to create the adoption, and either follow redirect or get the same page, and assert it is no longer there. If that makes sense?

If we were not testing the PUT request in another test I think it makes sense to keep here, but seeing as you are testing it in the test to make sure the Create Adoption button exists, it is duplicated here. But, it's not wrong...I think it's debatable.

Reply:
Oh ok, I only added the status change PUT request in there because I thought it might be required in order to proceed with creating the adoption. I'll make all those changes, they sound great